core/src/main/scala/cats/Traverse.scala

```diff
@@ -37,6 +37,26 @@ import scala.annotation.implicitNotFound
    */
   def traverse[G[_]: Applicative, A, B](fa: F[A])(f: A => G[B]): G[F[B]]
 
+  /**
+   * Given a function which returns a G effect, thread this effect
+   * through the running of this function on all the values in F,
+   * returning an F[A] in a G context, ignoring the values
+   * returned by provided function.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> import cats.implicits._
+   * scala> import java.io.IOException
+   * scala> type IO[A] = Either[IOException, A]
+   * scala> def debug(msg: String): IO[Unit] = Right(())
+   * scala> List("1", "2", "3").traverseTap(debug)
+   * res1: IO[List[String]] = Right(List(1, 2, 3))
+   * }}}
+   */
+  def traverseTap[G[_]: Applicative, A, B](fa: F[A])(f: A => G[B]): G[F[A]] =
+    traverse(fa)(a => Applicative[G].as(f(a), a))
```

Comment:
Technically we could use `Functor[G].as`, but we have the applicative anyway, so it probably doesn't matter.

Reply:
Would that work even if `traverse` requires an `Applicative` for `G`?

Reply:
it'll work because Functor <:< Applicative